src/System.Slices/src/System/SpanEqualsExtensions.cs

```diff
@@ -0,0 +1,138 @@
+            for (int i = 0; i < first.Length; i++) {
+                if (first.GetItemWithoutBoundariesCheck(i) != second.GetItemWithoutBoundariesCheck(i)) { 
+                    return false; 
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether two spans are equal (byte-wise) by comparing the elements by using memcmp
+        /// </summary>
+        /// <param name="first">A span of shorts to compare to second.</param>
+        /// <param name="second">A span of shorts T to compare to first.</param>
+        public static bool SequenceEqual(this Span<short> first, Span<short> second)
+        {
+            if (first.Length != second.Length) { return false; }
+
+            // pinvoke calls are expensive (mostly due to memory pinning) so we call memcmp only for big slices
+            if (first.Length >= 512) { return MemoryUtils.MemCmp(first, second); }
+
+            // for small slices we just use simple loop
+            for (int i = 0; i < first.Length; i++) {
+                if (first.GetItemWithoutBoundariesCheck(i) != second.GetItemWithoutBoundariesCheck(i)) { 
+                    return false; 
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether two spans are equal (byte-wise) by comparing the elements by using memcmp
+        /// </summary>
+        /// <param name="first">A span of integers to compare to second.</param>
+        /// <param name="second">A span of integers T to compare to first.</param>
+        public static bool SequenceEqual(this Span<int> first, Span<int> second)
+        {
+            if (first.Length != second.Length) { return false; }
+
+            // pinvoke calls are expensive (mostly due to memory pinning) so we call memcmp only for big slices
+            if (first.Length >= 256) { return MemoryUtils.MemCmp(first, second); }
+
+            // for small slices we just use simple loop
+            for (int i = 0; i < first.Length; i++) {
+                if (first.GetItemWithoutBoundariesCheck(i) != second.GetItemWithoutBoundariesCheck(i)) { 
+                    return false; 
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether two spans are equal (byte-wise) by comparing the elements by using memcmp
+        /// </summary>
+        /// <param name="first">A span of doubles to compare to second.</param>
+        /// <param name="second">A span of doubles T to compare to first.</param>
+        public static bool SequenceEqual(this Span<double> first, Span<double> second)
+        {
+            if (first.Length != second.Length) { return false; }
+
+            // pinvoke calls are expensive (mostly due to memory pinning) so we call memcmp only for big slices
+            if (first.Length >= 256) { return MemoryUtils.MemCmp(first, second); }
+
```

Comment:
You are right, I should have not compare floating points in bitwise way. I have removed the SequenceEqual< double > extension method and added tests for that case.